src/jbuild_types.ml

```diff
@@ -678,6 +678,60 @@ module Rule = struct
   let ocamlyacc_vjs = ocamlyacc_v1
 end
 
+module Menhir = struct
+  type t =
+    { base : string option
+    ; flags : string list
+    ; modules: string list
+    }
+
+  let v1 =
+    record
+      (field_o "base" string >>= fun base ->
+       field "flags" (list string) ~default:[] >>= fun flags ->
```

Comment:
I tried doing that but that seems to entail to move all the rule generation to gen_rules.ml to expand the vars properly. Doesn't seem like a bad idea, but it would require to create a `| Menhir of ...` constructor and then handling it in gen_rules - so not a trivial amount of work. Would like to confirm that this is what you're asking here before doing it.